### ISSUE TITLE
new icon is not refreshed after update

### ISSUE THREAD

**Issue by pixelr0** [ubports/ubuntu-touch#471]
fairphone - stable 15.04 ota3
so yeah.. i've just changed the icon in the latest version of fishy shooter v0.201
after updating fishy shooter i see the dash refresh but it shows the old icon, same if i manually refresh the dash by dragging down the app scope. workaround is to restart the phone

**Comment by NeoTheThird** [comment]
I don't know if it makes sense to fix this, since unity-8-dash is deprecated in the latest unity8... https://github.com/ubports/unity8/issues/14

**Comment by NeoTheThird** [comment]
Does it update in the dock?

**Comment by pixelr0** [comment]
it does update in the launcher

**Comment by NeoTheThird** [comment]
Ok, since it's non-critical, let's leave it for now. Once we move to the latest unity8, it'lll resolve itself.

**Comment by dobey** [comment]
Actually this won't resolve itself with the latest unity8, as the app drawer doesn't have any refreshing at all now. There was the beginnings of work to add API in ubuntu-app-launch so the app drawer could be signaled when a new app was installed, or one removed, but it was not completed.

**Comment by NeoTheThird** [comment]
@dobey That's interesting. Do you know how the launcher handles this?

**Comment by dobey** [comment]
I'm not sure why the launcher would refresh the icon and the scope wouldn't. IIRC, it always worked for me in the past. I'm also not sure if open store is doing all the same things as the old store scope was doing, but with the old store a dbus signal causes the scope to refresh (which is the same as pull-to-refresh).

**Comment by NeoTheThird** [comment]
@dobey Yeah, the openstore does that as well. https://github.com/UbuntuOpenStore/openstore-app/blob/master/openstore/clickinstaller.cpp#L139

We'll have to try and reproduce this issue again...

**Comment by dobey** [comment]
@NeoTheThird Ah, no, it's only poking at the scope then. That'd be the issue. So I guess this is confirmed.

**Comment by dobey** [comment]
On the other hand, the report seems a bit backward for this. The store is refreshing the scope, but not the launcher.

I know how to fix this, but it's going to be a little more work than I expected, so I'll have a branch in a few days probably, when I can get some time to really look at it.

**Comment by UniversalSuperBox** [comment]
@dobey, will you continue on this issue? Has it already been fixed?

**Comment by dobey** [comment]
Oh no. I got busy with other stuff, and totally forgot about this issue.

**Comment by UniversalSuperBox** [comment]
Would you like to continue with it or is this back in the unassigned queue?
